docs/api.md

```diff
@@ -0,0 +1,264 @@
+
+```
+$ ./target/debug/cloud-hypervisor --api-socket /tmp/cloud-hypervisor.sock
+Cloud Hypervisor Guest
+    API server: /tmp/cloud-hypervisor.sock
+    vCPUs: 1
+    Memory: 512 MB
+    Kernel: None
+    Kernel cmdline:
+    Disk(s): None
+```
+
+##### Create a Virtual Machine
+
+We want to create a virtual machine with the following characteristics:
+
+* 4 vCPUs
+* 1 GB of RAM
+* 1 virtio based networking interface
+* Direct kernel boot from a custom 5.5.0 Linux kernel located at `/opt/kata/vmlinux-virtio-fs-virtio-iommu`
+* Using a Clear Linux image as its root filesystem
+
+```shell
+#!/bin/bash
+
+curl --unix-socket /tmp/cloud-hypervisor.sock -i \
+     -X PUT 'http://localhost/api/v1/vm.create'  \
+     -H 'Accept: application/json'               \
+     -H 'Content-Type: application/json'         \
+     -d '{
+         "cpus":{"boot_vcpus": 4, "max_vcpus": 4},
+         "kernel":{"path":"/opt/kata/vmlinux-virtio-fs-virtio-iommu"},
+         "cmdline":{"args":"console=hvc0 reboot=k panic=1 nomodules i8042.noaux i8042.nomux i8042.nopnp i8042.dumbkbd root=/dev/vda3"},
+         "disks":[{"path":"/opt/images/clear-30080-kvm.img"}],
+         "rng":{"src":"/dev/urandom"},
+         "net":[{"ip":"192.168.10.10", "mask":"255.255.255.0", "mac":"12:34:56:78:90:01"}]
+         }'
+```
+
+#### Boot a Virtual Machine
+
+Once the VM is created, we can boot it:
+
+```shell
+#!/bin/bash
+
+curl --unix-socket /tmp/cloud-hypervisor.sock -i -X PUT 'http://localhost/api/v1/vm.boot'
+```
+
+#### Dump a Virtual Machine Information
+
+We can fetch information about any VM, as soon as it's created:
+
+```shell
+#!/bin/bash
+
+curl --unix-socket /tmp/cloud-hypervisor.sock -i \
+     -X GET 'http://localhost/api/v1/vm.info' \
+     -H 'Accept: application/json'
+```
+
+#### Reboot a Virtual Machine
+
+We can reboot a VM that's already booted:
+
+```shell
+#!/bin/bash
+
+curl --unix-socket /tmp/cloud-hypervisor.sock -i -X PUT 'http://localhost/api/v1/vm.reboot'
+```
+
+#### Shut a Virtual Machine Down
+
+Once booted, we can shut a VM down from the REST API:
+
+```shell
+#!/bin/bash
+
+curl --unix-socket /tmp/cloud-hypervisor.sock -i -X PUT 'http://localhost/api/v1/vm.shutdown'
+```
+
+### Command Line Interface
+
+The Cloud Hypervisor Command Line Interface (CLI) can only be used for launching
+the Cloud Hypevisor binary, i.e. it can not be used for controlling the VMM or
+the launched VM once they're up and running.
+
+If you want to inspect the VMM, or control the VM after launching Cloud
+Hypervisor from the CLI, you must use the [REST API](#rest-api).
+
+From the CLI, one can either:
+
+1. Create and boot a complete virtual machine by using the CLI options to build
+   the VM config. Run `cloud-hypervisor --help` for a complete list of CLI
+   options. As soon as the `cloud-hypervisor` binary is launched, the
+   [REST API](#rest-api) is available for controlling and managing the VM.
+1. Only start the [REST API](#rest-api) server when not passing any VM
```

Comment:
Following the "one can either:" sentence before, I would replace `Only start` with `Start`.

Reply:
Have the similar feeling, and think of the following alternative phrase:
"Start the [REST API](#rest-api) server only by not passing any VM ..."